src/scripts/main.js

```diff
@@ -2,10 +2,23 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, function handler() {
+      const message = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
+
+      resolve(message);
+      element.removeEventListener(eventName, handler);
+    });
+  });
 }
 
 const printMessage = (message) => {
   // write your code here
+  const messageDiv = document.createElement('div');
+
+  messageDiv.classList.add('message');
+  messageDiv.textContent = message;
+  document.body.appendChild(messageDiv);
 };
```

Comment:
Code quality issue: Your `printMessage` function is not only printing a message, but also creating the message container (`div`). It would be better to separate these responsibilities. You could create a function that creates and returns a `div` with a message, and another function that appends this `div` to the body.